VSCodeExtension/out/src/cncList.js

```diff
@@ -1,199 +1,199 @@
-    }
-    return cncFiles;
-}
-
-function getFiles(srcpath) {
-    return fs.readdirSync(srcpath);
-}
+"use strict";
+
+/*
+  Copyright (c) 2022 by Autodesk, Inc.
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+*/
+
+Object.defineProperty(exports, "__esModule", {value: true});
+const vsc = require("vscode");
+const fs = require("fs");
+const path = require("path");
+const exten = require("./extension");
+let resLocation = path.join(vsc.extensions.getExtension("Autodesk.hsm-post-processor").extensionPath, "res");
+let additionalFolders = path.join(resLocation, "CNC files", "customLocations.json");
+/** Contains a list of all available CNC files */
+let files = [];
+let allFilesName = "All files";
+
+class cncDataProvider {
+    constructor(_context) {
+        this._context = _context;
+        this._onDidChangeTreeData = new vsc.EventEmitter();
+        this.onDidChangeTreeData = this._onDidChangeTreeData.event;
+        files = [];
+        // find all the files in the default storage location
+        files = findFiles(path.join(resLocation, "CNC files"));
+        // find all the files defined in the additional folders
+        if (fs.existsSync(additionalFolders)) {
+            let lines = fs.readFileSync(additionalFolders);
+            if (lines.length > 1) {
+                let lineJson = JSON.parse(lines);
+                for (let i = 0; i < lineJson.folders.length; ++i) {
+                    if (fs.existsSync(lineJson.folders[i])) {
+                        files = files.concat([[path.basename(lineJson.folders[i]),lineJson.folders[i]]]);   
+                    }
+                }
+            }
+        }
+        files.unshift(allFilesName);
+    }
+    /** Gets the children of the selected item */
+    getChildren(element) {
+        let items = [];
+        if (!element) {
+            for (let i = 0; i < files.length; ++i) {

```

Comment:
One thing that would aid readability would be to extract some well named functions. This would also have the benefit of  reducing the level of nesting. It's been suggested that having more than 3 levels of nesting makes the code difficult to read.

Reply:
Good point @DMaffAD! I'll look into doing that